lib/customisation.nix

```diff
@@ -131,7 +131,10 @@ rec {
       origArgs = auto // args;
       pkgs = f origArgs;
       mkAttrOverridable = name: _: makeOverridable (newArgs: (f newArgs).${name}) origArgs;
-    in lib.mapAttrs mkAttrOverridable pkgs;
+    in
+      if lib.isDerivation pkgs then throw
+        "function callPackages called on a *single* derivation `${pkgs.name or "<unknown-name>"}`."
```

Comment:
I suggest this message instead: "`callPackages` was called on a single derivation `${pkgs.name or "<unknown-name>"}`, but it should be called on a set of derivations instead"

Reply:
Or the other way, we may suggest using `callPackage` instead.  I'd expect that to be the action to take in most cases.

Reply:
Sounds good, I suggest this then: "`callPackages` was called on the single derivation `${pkgs.name or "<unknown-name>"}`, did you mean to use `callPackage` instead?"

I think good error messages are very important, they can make a big difference to user experience.

Reply:
Applied.